crates/ide/src/inlay_hints.rs

```diff
@@ -353,13 +353,18 @@ fn is_argument_similar_to_param_name(
     }
     match get_string_representation(argument) {
         None => false,
-        Some(mut repr) => {
-            let param_name = param_name.to_ascii_lowercase();
-            let argument_string = {
-                repr.make_ascii_lowercase();
-                repr.trim_start_matches('_')
-            };
-            argument_string.starts_with(&param_name) || argument_string.ends_with(&param_name)
+        Some(argument_string) => {
+            let mut arg_bytes = argument_string.bytes().skip_while(|&c| c == b'_');
+            let starts_with_pattern = param_name.bytes().all(
+                |expected| matches!(arg_bytes.next(), Some(actual) if expected.eq_ignore_ascii_case(&actual)),
+            );
+
+            let mut arg_bytes = argument_string.bytes();
```

Comment:
Hm, this changes old behavior a bit due to `_` present in reverse search now.
Might make sense to omit underscores here too.

Ideally, we can come up with another test for this, but up to you.

Reply:
I thought about it, but decided against it. Given that only _leading_ underscores were stripped and the fact that the "ends with" does it in reverse, the only time this would make a difference is if the argument name had a trailing underscore. Needless to say I didn't think that was a common enough case to warrant consideration.

Either way, it's trivial to change if you'd like, and would presumably have minimal overhead.

Reply:
I was speculating mostly, thinking about the cases with the leading underscoders where it can matter:

```
argument_string = "_value" 
param_name      = "lvalue"
```

(horrible example, I'm too drowsy now to come up with the proper ones ad hoc, sorry)

I can imagine some autogenerated/bindings code with this issue, but maybe I'm just too pedantic here.

Reply:
Meh, can't hurt. It's a pretty small change, and adds little overhead over what's already there.

Oddly enough, manually counting the number of underscores and skipping it in later iterators is _far_ more efficient than using `.trim_start_matches()`. On the order of a 3 lines of assembly difference for what I ended up doing versus a 60+ line difference using the built-in method.

Reply:
Curious indeed, maybe @matklad can use this knowledge somewhere, who knows.

Thanks for looking into this.